js_modules/dagster-ui/packages/ui-core/src/automation/AutomationStateChangeDialog.tsx

```diff
@@ -0,0 +1,361 @@
+import {useMutation} from '@apollo/client';
+// eslint-disable-next-line no-restricted-imports
+import {ProgressBar} from '@blueprintjs/core';
+import {
+  Button,
+  Colors,
+  Dialog,
+  DialogBody,
+  DialogFooter,
+  Group,
+  Icon,
+} from '@dagster-io/ui-components';
+import {useEffect} from 'react';
+
+import {assertUnreachable} from '../app/Util';
+import {
+  OpenWithIntent,
+  useInstigationStateReducer,
+} from '../instigation/useInstigationStateReducer';
+import {BasicInstigationStateFragment} from '../overview/types/BasicInstigationStateFragment.types';
+import {NavigationBlock} from '../runs/NavigationBlock';
+import {START_SCHEDULE_MUTATION, STOP_SCHEDULE_MUTATION} from '../schedules/ScheduleMutations';
+import {
+  StartThisScheduleMutation,
+  StartThisScheduleMutationVariables,
+  StopScheduleMutation,
+  StopScheduleMutationVariables,
+} from '../schedules/types/ScheduleMutations.types';
+import {START_SENSOR_MUTATION, STOP_SENSOR_MUTATION} from '../sensors/SensorMutations';
+import {
+  StartSensorMutation,
+  StartSensorMutationVariables,
+  StopRunningSensorMutation,
+  StopRunningSensorMutationVariables,
+} from '../sensors/types/SensorMutations.types';
+import {RepoAddress} from '../workspace/types';
+
+export interface AutomationInfo {
+  repoAddress: RepoAddress;
+  name: string;
+  type: 'sensor' | 'schedule';
+  instigationState: BasicInstigationStateFragment;
+}
+
+export interface Props {
+  openWithIntent: OpenWithIntent;
+  onClose: () => void;
+  onComplete: () => void;
+  automations: AutomationInfo[];
+}
+
+export const AutomationStateChangeDialog = (props: Props) => {
+  const {openWithIntent, onClose, onComplete, automations} = props;
+  const count = automations.length;
+
+  const [state, dispatch] = useInstigationStateReducer();
+
+  // If the dialog is newly closed, reset state.
+  useEffect(() => {
+    if (openWithIntent === 'not-open') {
+      dispatch({type: 'reset'});
+    }
+  }, [openWithIntent, dispatch]);
```

Comment:
You can avoid this by moving this logic to a AutomationStateChangeDialogBody component. That way Blueprint creates a brand new instance when it's opened/closed.